File: miracle-api/src/main/java/com/depromeet/service/record/dto/response/RecordResponse.java
package com.depromeet.service.record.dto.response;

import com.depromeet.domain.common.Category;
import com.depromeet.domain.record.Record;
import lombok.Getter;
import lombok.RequiredArgsConstructor;
import lombok.ToString;

import java.time.LocalDateTime;

@ToString
@Getter
@RequiredArgsConstructor
public class RecordResponse {

    private final Long id;

    private final Long memberId;

    private final Long scheduleId;

    private final LocalDateTime startDateTime;

    private final LocalDateTime endDateTime;

    private final Category category;

    private final String question;

    private final String answer;

    public static RecordResponse of(Record record) {
        return new RecordResponse(record.getId(), record.getMemberId(), record.getScheduleId(), record.getStartDateTime(),
            record.getEndDateTime(), record.getCategory(), record.getQuestion(), record.getAnswer());
    }

}

File: miracle-api/src/main/java/com/depromeet/service/record/RecordResponse.kt
package com.depromeet.service.record

import com.depromeet.domain.common.Category
import com.depromeet.domain.record.Record
import java.time.LocalDateTime
import javax.persistence.EnumType
import javax.persistence.Enumerated

data class RecordResponse(
    val scheduleId: Long? = null,
    val startDateTime: LocalDateTime? = null,
    val endDateTime: LocalDateTime? = null,
    @field:Enumerated(EnumType.STRING)
    val category: Category? = null,
    val question: String? = null,
    val answer: String? = null
) {

    companion object {
        fun of(record: Record): RecordResponse {
            return RecordResponse(
                scheduleId = record.scheduleId,
                startDateTime = record.startDateTime,
                endDateTime =  record.endDateTime,
                category = record.category,
                question = record.question,
                answer = record.answer
            )
        }
    }
}



